src/relayServerUtils.ts

```diff
@@ -35,43 +39,99 @@ async function calculateFee(
   relayRequest: EnvelopingRequest,
   maxPossibleGas: BigNumber,
   appConfig: AppConfig
-) {
+): Promise<BigNumberJs> {
+  if (isSponsorshipAllowed(relayRequest, appConfig)) {
+    return BigNumberJs(0);
+  }
+
+  const { fixedUsdFee } = appConfig;
+  let totalFee = BigNumberJs(0);
+
+  if (fixedUsdFee) {
+    totalFee = await calculateFixedUsdFee(relayRequest, fixedUsdFee);
+
+    log.debug(
+      `Fixed USD fee applied. Value in USD: ${fixedUsdFee}, fee in gas: ${totalFee.toString()}`
+    );
+  }
+
   const { transferFeePercentage } = appConfig;
 
+  //Even if transferFeePercentage = 0, it has priority over gas fee
   if (
     transferFeePercentage >= 0 &&
     isTransferOrTransferFrom(relayRequest.request.data.toString())
   ) {
-    const feeInToken = calculateFeeFromTransfer(
+    const transferFee = await calculateFeeFromTransfer(
       relayRequest.request.data.toString(),
-      transferFeePercentage
+      transferFeePercentage,
+      relayRequest
     );
+
     log.debug(
-      `Transfer fee applied. Fraction: ${transferFeePercentage}, fee in tokens: ${feeInToken.toString()}`
+      `Transfer fee applied. Fraction: ${transferFeePercentage}, value in gas: ${transferFee.toString()}`
     );
 
-    return await convertTokenToGas(
-      feeInToken.toString(),
-      relayRequest.request.tokenContract.toString(),
-      relayRequest.relayData.gasPrice.toString()
-    );
+    return totalFee.plus(transferFee);
   }
 
   const { gasFeePercentage } = appConfig;
-  log.debug(`Gas fee applied. Fraction: ${gasFeePercentage}`);
 
-  return calculateFeeFromGas(maxPossibleGas.toString(), gasFeePercentage);
+  if (gasFeePercentage) {
+    const gasFee = calculateFeeFromGas(
+      maxPossibleGas.toString(),
+      gasFeePercentage
+    );
+    log.debug(
+      `Gas fee applied. Fraction: ${gasFeePercentage}, value in gas: ${gasFee.toString()}`
+    );
+
+    return totalFee.plus(gasFee);
+  }
+
+  return totalFee;
+}
+
+async function calculateFixedUsdFee(
+  envelopingRequest: EnvelopingRequest,
+  fixedUsdFee: number
+) {
+  const tokenAddress = envelopingRequest.request.tokenContract.toString();
+  const gasPrice = envelopingRequest.relayData.gasPrice.toString();
+
+  const provider = getProvider();
+
+  const tokenInstance = ERC20__factory.connect(tokenAddress, provider);
+  const tokenSymbol = await tokenInstance.symbol();
+
+  const relayPricer = new RelayPricer();
+  const exchange = relayPricer.findAvailableApi(tokenSymbol);
+  const inverseExchangeRate = await exchange.queryExchangeRate(
+    tokenSymbol,
+    US_DOLLAR_SYMBOL
+  );
+  const exchangeRate = BigNumberJs(1).dividedBy(inverseExchangeRate);
```

Comment:
what do we need to get the inverseExchangeRate before? what if we invert the arguments of `exchange.queryExchangeRate` to be:
```
await exchange.queryExchangeRate(
   US_DOLLAR_SYMBOL,
   tokenSymbol
  );
```

Reply:
I double checked this. We can not send USD as first parameter because of [this](https://github.com/rsksmart/rif-relay-client/blob/main/src/api/pricer/TestExchange.ts#L6) configuration. We could solve it by changing the queryExchangeRate() on that class or by adding the inverse configuration but in the second case we had to maintain two inverse/equivalent values. Probably for [this](https://github.com/rsksmart/rif-relay-client/blob/main/src/api/pricer/CoinBase.ts#L29) case the order of the currencies doesn't matter but for testing scenarios it will fail.